airflow/operators/redshift_to_s3_operator.py

```diff
@@ -0,0 +1,105 @@
+    """
+    Executes an UNLOAD command to s3 as a CSV with headers
+    :param schema: reference to a specific schema in redshift database
+    :type schema: string
+    :param table: reference to a specific table in redshift database
+    :type table: string
+    :param s3_bucket: reference to a specific S3 bucket
+    :type s3_bucket: string
+    :param s3_key: reference to a specific S3 key
+    :type s3_key: string
+    :param redshift_conn_id: reference to a specific redshift database
+    :type redshift_conn_id: string
+    :param s3_conn_id: reference to a specific S3 connection
+    :type s3_conn_id: str
+    """
+
+    template_fields = ()
+    template_ext = ()
+    ui_color = '#ededed'
+
+    @apply_defaults
+    def __init__(
+            self,
+            schema,
+            table,
+            s3_bucket,
+            s3_key,
+            redshift_conn_id='redshift_default',
+            s3_conn_id='s3_default',
+            autocommit=False,
+            parameters=None,
+            *args, **kwargs):
+        super(RedshiftToS3Transfer, self).__init__(*args, **kwargs)
+        self.schema = schema
+        self.table = table
+        self.s3_bucket = s3_bucket
+        self.s3_key = s3_key
+        self.redshift_conn_id = redshift_conn_id
+        self.s3_conn_id = s3_conn_id
+        self.autocommit = autocommit
+        self.parameters = parameters
+
+    def execute(self, context):
+        self.hook = PostgresHook(postgres_conn_id=self.redshift_conn_id)
+        self.s3 = S3Hook(s3_conn_id=self.s3_conn_id)
+        a_key, s_key = self.s3.get_credentials()
+
+        logging.info("Retrieving headers from %s.%s..." % (self.schema, self.table))
+
+        columns_query = """SELECT column_name
+                            FROM information_schema.columns
+                            WHERE table_schema = '{0}'
+                            AND   table_name = '{1}'
+                            ORDER BY ordinal_position
+                        """.format(self.schema, self.table)
+
+        cursor = self.hook.get_conn().cursor()
+        cursor.execute(columns_query)
+        rows = cursor.fetchall()
+        columns = map(lambda row: row[0], rows)
+        column_names = (', ').join(map(lambda c: "\\'{0}\\'".format(c), columns))
+        column_castings = (', ').join(map(lambda c: "CAST({0} AS text) AS {0}".format(c),
+                                            columns))
+
+        unload_query = """
+                            UNLOAD ('SELECT {0}
+                            UNION ALL
+                            SELECT {1} FROM {2}.{3}')
+                            TO 's3://{4}/{5}/{3}_'
+                            with
+                            credentials 'aws_access_key_id={6};aws_secret_access_key={7}'
+                            DELIMITER AS ','
+                            ADDQUOTES
+                            NULL AS ''
+                            ALLOWOVERWRITE
+                            PARALLEL OFF;
```

Comment:
Hard coding `PARALLEL OFF` seems like a pretty opinionated design choice especially since it is inconsistent with defaults and Amazon's recommendations (http://docs.aws.amazon.com/redshift/latest/dg/r_UNLOAD.html).  That being said, I'm excited to see this PR and look forward to the functionality being added to Airflow.  Thanks for all you've done so far.